<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>743</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>300</y>
      <width>47</width>
      <height>14</height>
     </rect>
    </property>
    <property name="text">
     <string>P:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>300</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X(1≤x&lt;p):</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>300</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>K(1 &lt; K &lt; p - 1):</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="PEdit">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>290</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="ElGamal">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>231</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>ElGamal Cryptosystem </string>
    </property>
   </widget>
   <widget class="QTextEdit" name="XEdit">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>290</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="KEdit">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>290</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="plainTextEdit">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>100</y>
      <width>251</width>
      <height>141</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>80</y>
      <width>47</width>
      <height>14</height>
     </rect>
    </property>
    <property name="text">
     <string>Text:</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="resultEdit">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>370</y>
      <width>311</width>
      <height>141</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>350</y>
      <width>47</width>
      <height>14</height>
     </rect>
    </property>
    <property name="text">
     <string>Result:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="decypherButton">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>100</y>
      <width>121</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Decipher</string>
    </property>
   </widget>
   <widget class="QPushButton" name="encypherButton">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>100</y>
      <width>121</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Encipher</string>
    </property>
   </widget>
   <widget class="QComboBox" name="gComboBox">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>430</y>
      <width>60</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="GLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>410</y>
      <width>47</width>
      <height>14</height>
     </rect>
    </property>
    <property name="text">
     <string>G:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="findGButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>350</y>
      <width>121</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Find G</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>743</width>
     <height>18</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Файл</string>
    </property>
    <addaction name="openAction"/>
    <addaction name="openCipherAction"/>
    <addaction name="saveAction"/>
    <addaction name="saveCipherAction"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="openAction">
   <property name="text">
    <string>Открыть текст</string>
   </property>
  </action>
  <action name="saveAction">
   <property name="text">
    <string>Сохранить текст</string>
   </property>
  </action>
  <action name="saveCipherAction">
   <property name="text">
    <string>Сохранить шифр</string>
   </property>
  </action>
  <action name="openCipherAction">
   <property name="text">
    <string>Открыть шифр</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
